backend-config/account_association.go

```diff
@@ -21,6 +24,50 @@ func (c *ConfigT) processAccountAssociations() {
 	}
 }
 
+// getAccountDefinition returns a pointer to the account definition for the given account definition name.
+// If the account definition name is empty or the account definition doesn't exist, it returns nil.
+//
+// Parameters:
+//   - accountDefinitionName: The name of the account definition to look up
+//   - accountAssociationLogger: The logger instance to use for logging
+//
+// Returns:
+//   - *AccountDefinition: A pointer to the account definition or nil if not found
+func (c *ConfigT) getAccountDefinition(accountDefinitionName string, accountAssociationLogger logger.Logger) *AccountDefinition {
+	if accountDefinitionName == "" {
+		return nil
+	}
+	if accountDefinition, exists := c.AccountDefinitions[accountDefinitionName]; exists {
+		return &accountDefinition
+	}
+	// Log error if account definition not found
+	accountAssociationLogger.Errorn("Account definition not found in configured accountDefinitions")
```

Comment:
Shall we also log the `accountDefinitionName` without which this log doesn't give meaningful insight